executors/rawexec/volume.go

```diff
@@ -0,0 +1,274 @@
+//
+// Copyright (c) 2015 The heketi Authors
+//
+// This file is licensed to you under your choice of the GNU Lesser
+// General Public License, version 3 or any later version (LGPLv3 or
+// later), or the GNU General Public License, version 2 (GPLv2), in all
+// cases as published by the Free Software Foundation.
+//
+
+package rawexec
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"github.com/heketi/heketi/executors"
+	"github.com/lpabon/godbc"
+)
+
+func (s *RawExecutor) VolumeCreate(host string,
+	volume *executors.VolumeRequest) (*executors.Volume, error) {
+
+	godbc.Require(volume != nil)
+	godbc.Require(host != "")
+	godbc.Require(len(volume.Bricks) > 0)
+	godbc.Require(volume.Name != "")
+
+	cmd := fmt.Sprintf("gluster --mode=script volume create %v ", volume.Name)
```

Comment:
If volume is allocated in the root directory, it's a danger behavior, should be forbidden :u7981: